pre-processing/primod/coupled_model.py

```diff
@@ -0,0 +1,53 @@
+import abc
+from collections.abc import Sequence
+from pathlib import Path
+from typing import Any
+
+from primod.driver_coupling.driver_coupling_base import DriverCoupling
+
+
+class CoupledModel(abc.ABC):
+    coupling_list: Sequence[DriverCoupling]
```

Comment:
mypy probably made you do this, but `Sequence` for an actual variable feels weird to me.
I would use `Sequence` only for function parameters and actual types like `list` for variables, but if `mypy` is the reason you did this you can leave it as is.

Reply:
Yeah it's due to mypy, since it complains that lists are invariant.
To be fair, any sequence also actually works here given the logic, which is just a loop over the driver couplings.